pkg/madmin/info-commands.go

```diff
@@ -195,17 +195,17 @@ type Services struct {
 
 // Buckets contains the number of buckets
 type Buckets struct {
-	Count uint64 `json:"count,omitempty"`
+	Count *uint64 `json:"count,omitempty"`
```

Comment:
Why not just remove the omitempty instead?

Reply:
The values are fetched using the function `loadDataUsageFromBackend` and populated in the struct only when this function doesn't return an error. In case of error, the count cannot be set.

Reply:
Using pointers for integers is a very non idiomatic style for Go, if needed you should have an error string additionally that provides this detail or use UintMax upon error as an unexpected value.

Reply:
Will look into removing the `omitempty` + adding `Error` field

Reply:
@klauspost @harshavardhana I've updated the PR with following changes:

 - removed `omitempty` from the `count` fields
 - added an `error` field in each of these `count` fields
 - removed usage of pointers. Client can now rely on the `error` field to know if there was any error when fetching this data.
 
 PTAL